recipes/opencv/4.x/conanfile.py

```diff
@@ -732,7 +743,11 @@ def add_components(components):
                 # TODO: we should also define COMPONENTS names of each target for find_package() but not possible yet in CMakeDeps
                 #       see https://github.com/conan-io/conan/issues/10258
                 self.cpp_info.components[conan_component].set_property("cmake_target_name", cmake_target)
-                self.cpp_info.components[conan_component].libs = [lib_name]
+                if self.options.build_world and conan_component == "opencv_world":
+                    self.cpp_info.components[conan_component].libs = [lib_name]
+                elif not self.options.build_world:
+                    self.cpp_info.components[conan_component].libs = [lib_name]
+                    
```

Comment:
Are you sure that all other targets are defined in official OpenCV config file when world is built?

Reply:
That why I add this condition, opencv_world will combine all generated dll with only one dll which called opencv_world.dll

When world is disable, all dll will be created, tested on my side and working well.
I will recheck with the new version of the conanfile.py

Reply:
I know, but:
- all other components part of world should depend on opencv_world. They shouldn't have requires, system libs or frameworks themselves, everything should be aggregated in requires/system_libs/frameworks of opencv_world component
- 2 extra modules are never part of world: img_hash & sfm (and its static dependencies correspondence, multiview and numeric)
- ippiw is not part of world

Reply:
I've updated https://github.com/conan-io/conan-center-index/pull/15193 to add `world` option.